lib/hooks/UnaryExpression.js

```diff
@@ -9,7 +9,11 @@ exports.format = function UnaryExpression(node) {
   if (node.operator === 'delete') {
     _ws.limitAfter(node.startToken, 1);
     _br.limitBefore(node.startToken, 'DeleteOperator');
+
     var endToken = node.endToken;
+    if (_tk.isWs(endToken.next)) {
+      _ws.limitAfter(endToken, 0);
```

Comment:
I tried several approaches to fixing this issue, but all had some side effects.

This solution works for all the following:
```javascript
  delete this['bar']
  delete this['amet'];
  delete this['ipsum'];
```